### PR TITLE
Remove no-op cancellations

### PR DESCRIPTION
**Pull request by matklad**
### What does this PR do?

`JoinHandle`s implement structured concurrency and already do
cancellation in drop, so there's no need for explicit cancellations.

### Additional Notes

One interesting thing I've noticed is that both `.cancel` and `.drop` are sync. Ideally, you'd `.await` for task after its cancellation. I am not sure how this should interact with scipio:

* on the one hand, because executor runs on a single thread, I *think* we can implement fully synchronous task cancellation, such that `cancel` immediately frees all task's resources. 
* on the other hand, the cancellation *should* be fully async. Let's say we have a task, which is waiting on an uring operation. Currently, when we cancel the task, we just enqueue cancellation operation into uring and, IIUC, we never actually await for the cancellation to happen. So, if you allocate 10GB buffer to do a read, and then cancel the read, that task will return from `.await` reporting that the cancellation concluded, but there still be 10GB buffer in the SourceMap for some time. 

### Checklist

[] I have added unit tests to the code I am submitting
[] My unit tests cover both failure and success scenarios
[] If applicable, I have discussed my architecture
[x] The new code I am adding is formatted using `rustfmt`

### CODE REVIEW

**Comment by matklad** [comment]
Wait, we detach the tasks, so this is wrong...

**Comment by matklad** [comment]
Yeah, so this PR is clearly wrong, because we are detaching the tasks. But I wonder *why* do we need to detach tasks here? Without having read the code to closely, it seems like tasks are scoped to `Stream`s, so it seems like a perfect use-case for drop-based emergency resource cleanup?

**Comment by glommer** [comment]
The whole purpose of the write-behind / read-ahead mechanism is that you will keep many such requests in flight.
They come and go as they please and you never wait for all of them together.

My understanding is that if I don't `detach` those tasks, I will have to `.await` them right away because rust futures are lazy and then it becomes quite hard to achieve this level of parallelism.

I could be wrong in that understanding, but in fact removing the `.detach` the compiler tells me this:

```
   Compiling scipio v0.1.0 (/home/glauber/scipio/scipio)
warning: unused `executor::Task` that must be used
   --> scipio/src/io/dma_file_stream.rs:714:9
    |
714 | /         Local::local(async move {
715 | |             defer! {
716 | |                 waker.wake();
717 | |             }
...   |
745 | |             drop(state);
746 | |         });
    | |___________^
    |
    = note: `#[warn(unused_must_use)]` on by default
    = note: tasks get canceled when dropped, use `.detach()` to run them in the background
```

**Comment by glommer** [comment]
@matklad are you still pursuing this after my explanation ?

**Comment by matklad** [comment]
Not at the moment: life happened :) I still think that there's a fair chance that non-detaching would work, but let's close this PR for now, as I don't have an estimate for when I can get back to it!
